File: javascript/pangram/pangram.js
/** INPUT: sentence (string)
*   OUTPUT: true if sentence is a pangram, i.e. it contains
              every letter of the English alphabet at least once. */
export const isPangram = (sentence) => {

  const letters_found = new Set(
    sentence.toLowerCase().match(/[a-z]/g));

  return letters_found.size === 26;

};

File: cpp/pangram/pangram.cpp
#include "pangram.h"
#include <bitset>
#include <cctype>

namespace pangram {

	bool is_pangram(const std::string& sentence)
	{
		// letters_found[i] is true (set) when the ith letter (0-based) of the English alphabet has been found in sentence, false otherwise.
		std::bitset<26> letters_found;

		for (auto c : sentence)
		{
			if (std::isalpha(c))
			{
				int alpha_index = std::tolower(c) - 'a';
				letters_found.set(alpha_index);
			}
		}

		return letters_found.all();
	}

}  // namespace pangram
